src/adapter/src/coord/message_handler.rs

```diff
@@ -306,12 +308,25 @@ impl Coordinator {
             self.get_local_write_ts().await.timestamp.into()
         };
 
-        let mut ops = vec![];
+        let mut ops = Vec::with_capacity(shards_usage.by_shard.len());
         for (shard_id, shard_usage) in shards_usage.by_shard {
-            ops.push(catalog::Op::UpdateStorageUsage {
-                shard_id: Some(shard_id.to_string()),
-                size_bytes: shard_usage.size_bytes(),
+            let id = match self.catalog().allocate_storage_usage_id().await {
+                Ok(id) => id,
+                Err(err) => {
+                    tracing::warn!("Failed to allocate ID for storage metrics: {:?}", err);
+                    return;
```

Comment:
This error path is surprising to me, but matches what we do if the catalog transaction fails.